support-frontend/assets/helpers/forms/formValidation.ts

```diff
@@ -26,52 +26,73 @@ import type { LocalCurrencyCountry } from '../internationalisation/localCurrency
 
 export const emailRegexPattern =
 	"^[a-zA-Z0-9\\.!#$%&'*+/=?^_`{|}~-]+@[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?(?:\\.[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?)+$";
+
 export const isEmpty: (arg0: string | null) => boolean = (input) =>
 	typeof input === 'undefined' || input == null || input.trim().length === 0;
+
 export const isNotEmpty: (arg0: string | null) => boolean = (input) =>
 	!isEmpty(input);
+
 export const isNotTooFarInTheFuture: (arg0: Date | null) => boolean = (
 	date,
 ) => {
 	const rangeDate = new Date();
 	rangeDate.setDate(rangeDate.getDate() + daysFromNowForGift);
+
+	if (!date) {
+		return false;
```

Comment:
Is there anywhere we call `isNotTooFarInTheFuture` with `null` currently? It feels like the type signature should prevent this and only allow a `Date` to be passed as a non-optional argument?

Reply:
`isNotTooFarInTheFuture` is only called by [`checkGiftStartDate`](https://github.com/guardian/support-frontend/blob/main/support-frontend/assets/helpers/forms/formValidation.ts#L69) which can be refactored